libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -754,6 +754,53 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		}
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (auto const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += (constructorParams.empty() ? "" : ", ") + IRVariable{*arg}.commaSeparatedList();
```

Comment:
I think there is the possibility of `commaSeparatedList()` to be empty, for example for string literals.

Reply:
Also you can remove the empty-check here and the `<comma>` below at the same time and make everything simpler :)

Reply:
Great observation :)